internal/extsvc/auth/auth.go

```diff
@@ -27,6 +27,21 @@ type Authenticator interface {
 	Hash() string
 }
 
+type AuthenticatorWithRefresh interface {
+	// Must implement the base Authenticator interface
+	Authenticator
+
+	// ShouldRefresh returns true if the Authenticator is no longer valid and
+	// needs to be refreshed, such as checking if an OAuth token is close to
+	// expiry or already expired.
+	ShouldRefresh() bool
```

Comment:
I know we had this discussion already and I swear this comment is non-blocking and you can print it and burn it and ignore it, *but*: how about... 

```suggestion
	NeedsRefresh() bool
```

IMHO this reads better.

```go
if ok := authenticator.NeedsRefresh(); ok {

}
```

since it fits into a sentence where the authenticator is the subject. `ShouldRefresh` IMHO means that the user of the authenticator should refresh, but if we think of methods such as `IsValid()` I think the pattern is that the method receiver is the subject in the sentence.

Again: I KNOW this is a nitpick, but can't help myself

Reply:
Hey man I'm all for picking some nits